src/functions/game/onPlayedTileInteraction.ts

```diff
@@ -0,0 +1,139 @@
+    }
+
+    /**
+     * Making Consecutive
+     */
+    // Only one user (the next user to the user who played the tile) can make consecutive
+    if (meld === MeldEnum.CONSECUTIVE) {
+      numOfConsecutive += 1;
+      connectionId = cid;
+    }
+
+    // If no one make a Triplet or Quad, then a user can make consecutive if there is any
+    if (!messageSent && meld === MeldEnum.CONSECUTIVE && numOfConsecutive === 1) {
+      const msg = createInteractionSuccessResponse({
+        ...wsPayload,
+        connectionId,
+      });
+      return Promise.all(connectionIds.map((user) => ws.send(msg, user)));
+    }
+
+    return wsPayload;
+  });
+};
+
+export const handler: Handler = async (event: WebSocketAPIGatewayEvent): Promise<LambdaResponse> => {
+  Logger.createLogTitle('onPlayedTileInteraction.ts');
+
+  // Parse event
+  const { connectionId } = event.requestContext;
+  const body: LambdaEventBody = JSON.parse(event.body);
+  const { payload }: { payload: LambdaEventBodyPayloadOptions } = body;
+  const gameId = payload.gameId as string;
+  const playedTile = payload.playedTile as string;
+  const meldType = payload.meldType as string;
+  const skipInteraction = payload.skipInteraction as boolean;
+
+  console.log('Incrementing interaction count and decide which meld type takes priority');
+  const ws = new WebSocketClient(event.requestContext);
+  const playedTileResponse = { playedTile, meldType, skipInteraction };
+  const playedTileInteractionResponse = createPlayedTileInteractionResponse(playedTileResponse);
+  try {
+    // Get current interaction count
+    const interactionCount = (await getInteractionCount(gameId)) as number;
+
+    // Run setPlayedTileInteraction when interaction count is less than 4
+    let newGameState: GameState;
+    let newInteractionCount = 0;
+    if (interactionCount < DEFAULT_MAX_USERS_IN_GAME - 1) {
+      newGameState = (await setPlayedTileInteraction(
+        gameId,
+        connectionId,
+        playedTile,
+        meldType,
+        skipInteraction,
+      )) as GameState;
+      newInteractionCount = newGameState.interactionCount as number;
+      await ws.send(successWebSocketResponse(playedTileInteractionResponse), connectionId);
+    }
+
+    // Compare meld type and send message based on priority
+    console.log('newInteractionCount:', newInteractionCount);
+    let interactionEnded = false;
+    if (newInteractionCount === 3) {
+      interactionEnded = true;
+      await compareTileInteractionAndSendUpdate(gameId, ws);
+    }
+
+    // Reset interactionCount to be 0 and playedTile list to empty
+    if (interactionEnded) {
+      await resetPlayedTileInteraction(gameId);
+    }
+
+    return response(200, 'Tile interaction is successful');
+  } catch (err) {
+    console.error(JSON.stringify(err));
+    await ws.send(failedWebSocketResponse(playedTileInteractionResponse, err), connectionId);
+    return response(500, err);
+  }
+};
```

Comment:
Very nice. This lambda was implemented based on what I had in mind

Reply:
thx, need more manual testing though haha